### PR TITLE
Bump runtime to 42

### PR DESCRIPTION
**Pull request by bbhtt**
Ping @adnan449, please review when you have time. Thanks!

Build fails with Nim 1.6.6, opened an issue about this.

### CODE REVIEW

**Comment by bbhtt** [comment]
bot, build

File: com.gitlab.adnan338.Invoicer.yaml [hunk 1]
```diff
@@ -20,13 +20,13 @@ modules:
       - "*"
     build-commands:
       - >
-          if [ $(uname -m) == "x86_64" ]; then 
+          if [ $(uname -m) == "x86_64" ]; then
```
Comment (hfiguiere) [inline com.gitlab.adnan338.Invoicer.yaml]: you could also change this (and below) to have `only-arches`. This would save a lot of downloads.

Comment (bbhtt) [reply com.gitlab.adnan338.Invoicer.yaml]: Thanks, done.

**Comment by bbhtt** [comment]
> Started [test build 101696](https://buildbot.flathub.org/#/builders/32/builds/101696)

Build is fine, issue with flathub servers so uploading fails :(

`Api call to https://hub.flathub.org/api/v1/build/99415/build_ref failed with status 500, details: {'status': 500, 'error-type': 'no-error', 'message': 'No json error details from server'}`

**Comment by hfiguiere** [comment]
retriggered

**Pull request merged**